### PR TITLE
TECH-517: Ensure DBAS Docker image is production ready

### PR DESCRIPTION
**Pull request by hbielenia**
This PR makes DBAS more conformant with the production checklist at https://assemblyvoting.atlassian.net/browse/TECH-434?focusedCommentId=12354

`build-and-push` job has been refactored and `scan` should be now marked required instead of it.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,83 +5,137 @@ orbs:
...
+            - docker/ssh-access:
+                VAR_SSH_KEY: PULL_SSH_KEY
+            - run:
+                name: Build Docker image
+                command: |
+                  docker build . -t dbas:ssh-${CIRCLE_SHA1} --ssh default --target production
+            - run:
+                name: Push Docker image
+                command: |
+                  docker tag dbas:ssh-${CIRCLE_SHA1} ${ECR_REGISTRY}/dbas:ssh-${CIRCLE_SHA1}
```
Comment (av-sebastian) [inline .circleci/config.yml:50]: @hbielenia why are the images tagged with prefix ss+?

**Comment by av-sebastian** [comment]
The latest production image in ECR should also have the tag "main" and "latest" for now. This is true for all projects excepts "Conference"

File: .circleci/config.yml (hunk 1 above)
Comment (hbielenia) [reply .circleci/config.yml]: I misunderstood the point of `docker/ssh-access` command, assuming it's there to give SSH access to container, which is undesirable in production image. But under closer examination, it looks to be giving the container an access to private repositories, so I scrapped the separate image in newest revision.

**Review by av-sebastian — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,26 @@
-FROM node:18-buster-slim
+# syntax=docker/dockerfile:1
```
[the remark this answers is not in the document]
Comment (hbielenia) [reply Dockerfile:1]: You mean if this affects editor settings? It's meant primary for Docker itself, I don't know if any editor makes use of it. My VSCode doesn't ;)

[the remark this answers is not in the document]
Comment (av-lukas) [reply Dockerfile:1]: I've never seen this before, hence the question.

**Pull request merged**